apps/tmbitwarden.sh

```diff
@@ -61,39 +60,165 @@ msg_box "OK, please do the necessary preparations before you run this script and
 
 To run this script again, execute $SCRIPTS/menu.sh and choose Additional Apps --> Bitwarden"
     exit
-else
-    sleep 0.1
 fi
 
+msg_box "IMPORTANT, PLEASE READ!
+
+In the next steps you will be asked to answer some questions.
+The questions are from the Bitwarden setup script, and therefore nothing that we control.
+
+It's important that you answer the questions correclty for the rest of the setup to work properly,
+and to be able to generate a valid TLS certificate automatically with our own (this) script.
+
+Basically:
+1. Enter the domain for Bitwarden
+2. Answer 'no' to the question if you want Let's Encrypt
+3. Enter your installation id and keys
+4. Continue to answer 'no' to everything related to SSL/TLS.
+
+Please have a look at how the questions are answered here if you are uncertain:
+https://i.imgur.com/YPynDAf.png"
+
 # Install Docker
 install_docker
 install_if_not docker-compose
 
-# Stop Apache to not conflict when LE is run
-check_command systemctl stop apache2.service
-
 # Install Bitwarden 
 install_if_not curl
 cd /root
 curl_to_dir "https://raw.githubusercontent.com/bitwarden/core/master/scripts" "bitwarden.sh" "/root"
 chmod +x /root/bitwarden.sh
 check_command ./bitwarden.sh install
-sed -i "s|http_port.*|http_port: 8080|g" /root/bwdata/config.yml
-sed -i "s|https_port.*|https_port: 8443|g" /root/bwdata/config.yml
+
+# Check if all ssl settings were entered correctly
+if grep ^url /root/bwdata/config.yml | grep -q https || grep ^url /root/bwdata/config.yml | grep -q localhost
+then
+    message "It seems like you have entered some wrong settings. We will remove bitwarden now again so that you can start over again."
+    check_command ./bitwarden.sh install
+    docker system prune -af
+    rm -rf /root/bwdata
+    exit 1
+fi
+
+# Continue with the installation
+sed -i "s|http_port.*|http_port: 5178|g" /root/bwdata/config.yml
+sed -i "s|https_port.*|https_port: 5179|g" /root/bwdata/config.yml
+# Get Subdomain from config.yml and change it to https
+SUBDOMAIN=$(grep ^url /root/bwdata/config.yml)
+SUBDOMAIN=${SUBDOMAIN##*url: http://}
+sed -i "s|^url: .*|url: https://$SUBDOMAIN|g" /root/bwdata/config.yml
+sed -i 's|http://|https://|g' /root/bwdata/env/global.override.env
 check_command ./bitwarden.sh rebuild
 check_command ./bitwarden.sh start
-if check_command ./bitwarden.sh updatedb
+check_command ./bitwarden.sh updatedb
```

Comment:
@enoch85 so this seems to make still problems. As I read in the bitwarden issue...

BTW: Why is it needed at all?
I couldn't find the reason why this is needed in the bitwarden install manual...

Reply:
Hm, Maybe this could help:
https://github.com/bitwarden/server/issues/155#issuecomment-381177257

Reply:
Read this: investigate and make a PR. Sorry, I'm busy now.

Thanks!